### ISSUE TITLE
attempting to reference a deleted function

### ISSUE THREAD

**Issue by Kobbersvard** [Equilibrium-Games/Acid#17]
Hi guys,

When on stage of building Acid.dll I see several errors like this (same stuff about deleted function) and build failed (VS 2017 15.7.6):

C:\Dev\Vulkan\Acid\Sources\Meshes\RendererMeshes.cpp(11): error C2280: 'acid::UniformHandler::UniformHandler(const acid::UniformHandler &)': attempting to reference a deleted function
  C:\Dev\Vulkan\Acid\Sources\Renderer/Handlers/UniformHandler.hpp(68): note: compiler has generated 'acid::UniformHandler::UniformHandler' here
  C:\Dev\Vulkan\Acid\Sources\Renderer/Handlers/UniformHandler.hpp(68): note: 'acid::UniformHandler::UniformHandler(const acid::UniformHandler &)': function was implicitly deleted because a data member invokes a deleted or inaccessible function 'std::unique_ptr<acid::UniformBuffer,std::default_delete<_Ty>>::unique_ptr(const std::unique_ptr<_Ty,std::default_delete<_Ty>> &)'
          with
          [
              _Ty=acid::UniformBuffer
          ]
  C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.14.26428\include\memory(2308): note: 'std::unique_ptr<acid::UniformBuffer,std::default_delete<_Ty>>::unique_ptr(const std::unique_ptr<_Ty,std::default_delete<_Ty>> &)': function was explicitly deleted
          with
          [
              _Ty=acid::UniformBuffer
          ]
  [10/86] C:\PROGRA~2\MICROS~1\2017\COMMUN~1\VC\Tools\MSVC\14.14.26428\bin\HostX86\x86\cl.exe  /nologo /TP -DACID_BUILD_DEBUG -DACID_BUILD_MSVC -DACID_BUILD_WINDOWS -DACID_EXPORTS -DACID_VERBOSE -DGLFW_DLL -DNOMINMAX -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS -IC:\Dev\Vulkan\Acid\Libraries\vulkan\include -IC:\Dev\Vulkan\Acid\Libraries\openal\include -IC:\Dev\Vulkan\Acid\Libraries\glslang -IC:\Dev\Vulkan\Acid\Libraries\glfw\include -IC:\Dev\Vulkan\Acid\Libraries\bullet3\src -IC:\Dev\Vulkan\Acid\Sources -IC:\Dev\Vulkan\Acid\Libraries\glslang\glslang\.. -IC:\Dev\Vulkan\Acid\Libraries\glslang\SPIRV\.. /DWIN32 /D_WINDOWS /W3 /GR /EHsc /DWIN32 /D_WINDOWS /std:c++17 /INCREMENTAL:NO /Oi /Ot /GT /GF /GS- /fp:fast /fp:except- /MP /W3 /wd4251 /arch:SSE2 /MDd /Zi /Ob0 /Od /RTC1 /Od /Ob0 /RTC1 /MDd /DEBUG /Zi /Od /Ob0 /RTC1 /MDd /DEBUG /Zi   -std:c++17 /showIncludes /FoSources\CMakeFiles\Acid.dir\Physics\ColliderCapsule.cpp.obj /FdSources\CMakeFiles\Acid.dir\ /FS -c C:\Dev\Vulkan\Acid\Sources\Physics\ColliderCapsule.cpp

**Comment by mattparks** [comment]
How exactly are you going about compiling Acid? I have testing building Debug and Release on Visual Studio 2017 15.8.3 without any issues. Are you using `Scripts/generate_vs.bat` or are you opening the project in Visual Studio to open it as a cmake project?
I can see that you are using MSVC version 14.14.26428, I have tested using 14.15.26726 (VC++ 2017 version 15.8 v14.15). Also I have only tested the 64bit build mode, if you are using 32bit then that could be an issue.
